docs/sql-guide/statements/statement-insert-bulk.md

```diff
@@ -0,0 +1,165 @@
+---
+title: BULK INSERT
+layout: default
+parent: Statements
+grand_parent: SQL guide
+---
+
+# BULK INSERT statement
+
+BULK INSERT data from CSV, NDJSON, URL or streaming sources to a FeatureBase table.
+
+## BNF diagrams
+
+![expr](/assets/images/sql-guide/bulk_insert_stmt.svg)
+
+### Column list
+![expr](/assets/images/sql-guide/column_list.svg)
+
+### Map list
+![expr](/assets/images/sql-guide/map_list.svg)
+
+### Transform clause
+![expr](/assets/images/sql-guide/map_list.svg)
+
+### With clause
+![expr](/assets/images/sql-guide/bulk_insert_options.svg)
+![expr](/assets/images/sql-guide/bulk_insert_option.svg)
+
+## DDL syntax
+
+```
+BULK [INSERT | REPLACE]
+  INTO
+    table_name [(column_name,...)]
+  MAP (position type_name,...)
+  [TRANSFORM (expr,...)]
+  FROM
+    ['path/file_name' | 'URL' | 'stream' ]
+  WITH
+    [
+      [BATCHSIZE integer_literal]
+      [ROWSLIMIT integer_literal]
+      [INPUT ['FILE' | 'URL' | 'STREAM']]
+      [FORMAT ['csv' [HEADER_ROW]] | ['ndjson' [ALLOW_MISSING_VALUES]]]
+      ...
+    ]
+```
+
+## Arguments
+
+| Argument | Description | Further information |
+|---|---|---|
+| `INSERT` | Update values in an existing row. |  |
+| `REPLACE` | Replace values in an existing row. |  |
+| `table_name` | Name of target table |  |
+| `column_name` | Valid columns belonging to `table_name`. First column must be defined `_id` column. System builds a column list from existing columns in `table_name` if columns are not specified. |  |
+| `MAP` | MAP defines how the source data is read and the expected data types. Values from the MAP clause are placed directly into the columns specified in the `column_list`. | [Map examples](/docs/sql-guide/statements/statement-insert-bulk/#map-examples)
+| `position` | Ordinal position of value in source. |  |
+| `type_name` | Data type of the value in source. |  |
+| `TRANSFORM expr` | a list of expressions that are evaluated during execution for each row. | [TRANSFORM examples](/docs/sql-guide/statements/statement-insert-bulk/#transform-clause-1) |
+| `FROM` | A single or multi-line string literal that specifies the source of data and are interpreted based on the INPUT option. Quoted string literals must be properly escaped. | [FROM quotation mark examples](/docs/sql-guide/statements/statement-insert-bulk/#from-quotation-examples) |
+| `'path/file_name'` | Valid path and file name for data source. Not available for FeatureBase Cloud. |  |
+| `'URL'` | Valid URL for data source. |  |
+| `'Inline_stream'` | The contents of the literal read as though they were in a file.  | [FROM inline stream examples](/docs/sql-guide/statements/statement-insert-bulk/#from-inline-stream) |
```

Comment:
you specify this with `x` so I'm not sure where inline_stream came from. Good to put in description though

Reply:
Sorry, I don't understand what you mean :(

Reply:
It took a bit of effort but I finally worked out where I'd gone wrong and have now fixed so everything actually matches what's in the syntax. >.<